### ISSUE TITLE
Bug: [no-multiple-lines-of-errors] adding `assertionOptions` disables the rule

### ISSUE THREAD

**Issue by Vinccool96** [typescript-eslint/typescript-eslint#12812]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Playground Link

NA

### Repro Code

```TypeScript
import { RuleTester } from '@typescript-eslint/rule-tester';

import rule from '../../src/rules/no-multiple-lines-of-errors.js';

const ruleTester = new RuleTester();

ruleTester.run('no-multiple-lines-of-errors', rule, {
  assertionOptions: {},
  invalid: [
    {
      code: `
ruleTester.run('test', rule, {
  invalid: [
    {
      errors: [
        { messageId: '...', line: 1 },
        { messageId: '...', line: 2 },
        { messageId: '...', line: 3 },
      ],
    },
  ],
});
      `,
      errors: [
        {
          column: 9,
          endColumn: 38,
          endLine: 7,
          line: 7,
          messageId: 'multipleLines',
        },
        {
          column: 9,
          endColumn: 38,
          endLine: 8,
          line: 8,
          messageId: 'multipleLines',
        },
      ],
    },
  ],
});
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/internal/no-multiple-lines-of-errors": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Multiple lines of error

### Actual Result

No multiple lines of error

**Comment by m-altaifi** [comment]
I looked into this. The cause is in the rule's `CallExpression` handler: it iterates the options object's properties and `return`s, rather than `continue`s, on the first property that is not `invalid`.

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin-internal/src/rules/no-multiple-lines-of-errors.ts#L82-L91

`assertionOptions` sorts before `invalid`, so it aborts the whole handler and nothing gets checked. Same for any file that puts `valid` before `invalid`.

Swapping `return` for `continue` fixes it, but it also turns the rule back on where it has been quietly off. Locally that is 479 reports across 61 of the 130 files matched by the `test-files/rules` config block, including `plugin-test-formatting.test.ts` in this package. They look like genuine violations rather than false positives, e.g. `no-implied-eval.test.ts` has one snippet asserting four errors on four lines.

That is more than a drive-by fix can reasonably split up, so I have the change staged with the 61 files grandfathered into an `off` block in `eslint.config.mjs`, to be removed as tests get split. Lint and the `eslint-plugin-internal` suite are green that way.

Happy to open the PR if this is worth `accepting prs`, and equally happy to shape it differently, such as leaving the rule off for `packages/eslint-plugin/tests/rules/**` entirely until the existing cases are cleaned up.

**Comment by kirkwaiblinger** [comment]
Lol. PR is welcome. Optional to fix violations or disable/suppress and fix them gradually as follow ons.
